comp/core/workloadmeta/def/types.go

```diff
@@ -1761,6 +1761,15 @@ func (p *Process) Merge(e Entity) error {
 		return fmt.Errorf("cannot merge ProcessMetadata with different kind %T", e)
 	}
 
+	// Preserve APMInstrumentation from destination if source has it as false
```

Comment:
I don't see what this fixes, the process collector [already](https://github.com/DataDog/datadog-agent/blob/vitkyrka/rust/comp/core/workloadmeta/collectors/internal/process/process_collector.go#L439-L442) preserves all static fields (not just `APMInstrumentation`) on heartbeat updates.